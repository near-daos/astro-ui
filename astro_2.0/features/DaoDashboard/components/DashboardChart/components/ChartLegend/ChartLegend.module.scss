@import 'styles/mixins';
@import 'styles/typography';

.root {
  align-items: center;
  display: flex;
  margin-right: 24px;
}

.legendIndicator {
  background-color: var(--color-neutral-50);
  border-radius: 50%;
  height: 8px;
  margin-right: 6px;
  width: 8px;
}

.legendLabel {
  @extend %title4;

  &::first-letter {
    text-transform: uppercase;
  }
}
