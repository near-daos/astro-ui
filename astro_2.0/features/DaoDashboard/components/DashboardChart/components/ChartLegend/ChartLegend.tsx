import React, { FC } from 'react';
import cn from 'classnames';

import styles from './ChartLegend.module.scss';

interface ChartLegendProps {
  label: string;
  className?: string;
}

export const ChartLegend: FC<ChartLegendProps> = ({ label, className }) => {
  return (
    <div className={styles.root}>
      <span className={cn(styles.legendIndicator, className)} />
      <span className={styles.legendLabel}>{label}</span>
    </div>
  );
};
