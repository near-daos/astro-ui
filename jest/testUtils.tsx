import {
  RenderResult,
  render as reactTestingLibRender,
} from '@testing-library/react';
import { ReactNode } from 'react';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';

function createMockRouter(router?: Partial<NextRouter>): NextRouter {
  return {
    basePath: '',
    pathname: '/',
    route: '/',
    query: {},
    asPath: '/',
    back: jest.fn(),
    beforePopState: jest.fn(),
    prefetch: jest.fn(),
    push: jest.fn(),
    reload: jest.fn(),
    replace: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isLocaleDomain: false,
    isReady: true,
    defaultLocale: 'en',
    domainLocales: [],
    isPreview: false,
    ...router,
  };
}

export function render(
  component: ReactNode,
  router?: Partial<NextRouter>
): RenderResult {
  return reactTestingLibRender(
    <RouterContext.Provider value={createMockRouter(router)}>
      {component}
    </RouterContext.Provider>
  );
}
