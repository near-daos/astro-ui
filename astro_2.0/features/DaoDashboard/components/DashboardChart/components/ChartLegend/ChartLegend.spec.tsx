import { render } from 'jest/testUtils';

import { ChartLegend } from 'astro_2.0/features/DaoDashboard/components/DashboardChart/components/ChartLegend/ChartLegend';

describe('ChartLegend', () => {
  it('Should render Chart legend component', () => {
    const { container } = render(
      <ChartLegend label="DAO funds" className="indicatorClassName" />
    );

    expect(container).toMatchSnapshot();
  });
});
