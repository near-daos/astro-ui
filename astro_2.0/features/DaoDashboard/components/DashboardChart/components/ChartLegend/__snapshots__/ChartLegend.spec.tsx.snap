// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartLegend Should render Chart legend component 1`] = `
<div>
  <div
    class="root"
  >
    <span
      class="legendIndicator indicatorClassName"
    />
    <span
      class="legendLabel"
    >
      DAO funds
    </span>
  </div>
</div>
`;
